core/indexer/indexer.go

```diff
@@ -162,6 +166,12 @@ func (i *indexer) Index(ctx context.Context, info smartblock.DocInfo, options ..
 	indexSetTime := time.Now()
 	var hasError bool
 	if indexLinks {
+		for _, link := range info.Links {
+			if strings.HasPrefix(link, addr.IdentityPrefix) && info.Space.DerivedIDs().Profile != "" {
```

Comment:
maybe this code should be placed where links are firstly populated?

Reply:
Yeah, moved it to getDocInfo